### DISCUSSION TITLE
Best way to load arbitrary (API) content?

### DISCUSSION THREAD

**Discussion by richardhj** [statamic/cms#7811] [Q&A] [answered]
What's the best way, to read arbitrary content (in the form of collection/entries) in the template engine?

More concretely, I want to use `{{ collection:events }}`, where the events collection shall be pulled from a remote API (Notion, in that case).
I've read the docs about _Repositories_ and _Data Retrieval and Manipulation_, but I need a little kickstart to get it working.

A custom route would be easy here, but I want to maintain CMS functionality.

Thanks!

**Comment by duncanmcclean** [comment]
I'd probably opt for building custom routes/controllers or building your own tag to get the API data and return it into your templates however you want.

**Comment by richardhj** [comment, marked as answer]
[Tags](https://statamic.dev/extending/tags) are awesome, thanks!
So I decided to go with Tags when I have content that repeats and with [custom routes](https://statamic.dev/controllers#routes) when the content is not repeated.
